internal/pkg/sysinfo/probes/filesystem_linux.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2024 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probes
+
+import (
+	"os"
+	"path"
+
+	"golang.org/x/sys/unix"
+)
+
+func (a *assertFileSystem) Probe(reporter Reporter) error {
+	var stat unix.Statfs_t
+	for p := a.fsPath; ; {
+		if err := unix.Statfs(p, &stat); err != nil {
+			if os.IsNotExist(err) {
+				if parent := path.Dir(p); parent != p {
+					p = parent
+					continue
+				}
+			}
+			return reporter.Error(a.desc(), err)
+		}
+		a.fsPath = p
+		break
+	}
+
+	var fs string
+	// https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/plain/include/uapi/linux/magic.h
+	switch stat.Type {
+	case unix.AAFS_MAGIC:
+		fs = "aafs" // 0x5a3c69f0
```

Comment:
Do you think the comments with the hex values are still useful, since we have the named constants now?

Reply:
I was thinking of using them to find duplicates, but the LSP did the job, so I think we can remove them.